### PR TITLE
Restaurant

### PR DESCRIPTION
**Pull request by po4erk**

### CODE REVIEW

**Comment by olbesp** [comment]
Cool!! We should change the font size of a Counter. It must be a little bit smaller. 
The Restaurant screen works fine.
screenshot from 2018-06-02 12-37-48

**Comment by olbesp** [comment]
We need a button 'Order Now' under the DishList. 
screenshot from 2018-06-02 21-37-57

File: src/components/Counter.js [hunk 1]
```diff
@@ -7,13 +7,17 @@ const styles = StyleSheet.create({
         borderWidth: 2,
         borderColor: 'lightgray',
         height: 35,
-        width: 85,
+        maxWidth: 85,
```
Comment (iarickvigasi) [inline src/components/Counter.js]: Why? 
What if a user will order 100 items? 
Will it break page layout?

File: src/components/DishList.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    fontWeight: '500',
+  },
+  subTitle: {
+    color: '#858b93',
+    fontSize: 20
+  }
+})
+
+const DishList = (props) => {
+    const item = props.items.map(item =>
```
Comment (iarickvigasi) [inline src/components/DishList.js]: Note for future 
Names like 'Item', 'Data' are not descriptive 
By reading variable Item we can't fastly understand what it is

File: src/screens/Restaurants/index.js [hunk 3]
```diff
@@ -66,7 +64,7 @@ class Restaurants extends React.Component<Props, State> {
 										style={styles.bannerImage}
 										imageUrl={getImageAddress(restaurant.photoUrl)}
 										id={restaurant.id}
-										onPress={this.restaurantCardPressHandler}
+										onPress={this.restaurantInfo.bind(this, restaurant)}
```
Comment (iarickvigasi) [inline src/screens/Restaurants/index.js]: You've declared restaurantInfo function as a fat arrow () => {} 
Which should bind context 
Doesn't it work? 
If not, think why)

File: src/components/Counter.js (hunk 1 above)
Comment (po4erk) [reply src/components/Counter.js]: does not break with 100+ items

File: src/screens/Restaurant/index.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	static navigationOptions = {
+		header: null,
+	}
+
+	state = {
+		dishesForOrder: [],
+		isShow: false
+	}
+
+	addDishesforOrder = (dish) => {
```
Comment (iarickvigasi) [inline src/screens/Restaurant/index.js]: camelCase

File: src/screens/Restaurant/index.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+			else{
+				allDishes.push(dish);
+			}
+		}
+		this.setState({
+			dishesForOrder: allDishes
+		})
+	}
+
+	showOrderBtn = () => {
```
Comment (iarickvigasi) [inline src/screens/Restaurant/index.js]: This logic will be removed after connecting to reducer

File: src/components/DishItem.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  }
+})
+
+class DishItem extends React.Component {
+
+    state = {
+        count: 0,
+    }
+
+    pushDishforOrder = () => {
```
[the remark this answers is not in the document]
Comment (po4erk) [reply src/components/DishItem.js]: fixed

File: src/components/DishItem.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        const dishId = this.props.dish.id
+        let dish = {dishId, count: this.state.count}
+        if(this.state.count != 0)
+            this.props.addDishesforOrder(dish)
```
[the remark this answers is not in the document]
Comment (po4erk) [reply src/components/DishItem.js]: fixed

File: src/screens/Restaurant/index.js (hunk 4 above)
Comment (po4erk) [reply src/screens/Restaurant/index.js]: fixed

**Pull request merged**